os/drivers/meminfo/meminfo_drv.c

```diff
@@ -98,7 +116,7 @@ static int heapinfo_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 			break;
 		}
 		if (heap == NULL) {
-			return ERROR;
+			return -EINVAL;
```

Comment:
I think if option->heap_type is HEAPINFO_HEAP_TYPE_KERNEL, heap is not NULL all the time, so this state had better move to line 112~113.